src/Couchbase.Lite.Shared/Serialization/FleeceMutableArray.cs

```diff
@@ -227,6 +261,34 @@ public bool Remove(object item)
 
         #endregion
 
+        #region IList<object>
+
+        public int IndexOf(object item)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Insert(int index, object val)
+        {
+            if (!IsMutable) {
```

Comment:
I wonder if `IsMutable` will ever be `false` anymore since it cannot store a non mutable array....Can you find a situation like that?

Reply:
I was asking in general, not just about this line.  If it is not used then the property itself can be removed but I have a feeling that somehow this property needs to protect against writing into an array that is supposed to be immutable.

Reply:
Looks like `IsMutable` is never be false... I don't see any public way to change the IsMutable to false.. However, the property is used in `public bool IsReadOnly => !IsMutable;` and IsReadOnly cannot be removed because it's the member of IList.

Reply:
It can get changed via `InitAsCopyOf` or `InitInSlot` from what I can tell so let's leave it in to be safe.